src/runtime/hexagon_remote/host_malloc.cpp

```diff
@@ -52,14 +79,31 @@ struct ion_handle_data {
 #define ION_IOC_FREE _IOWR('I', 1, ion_handle_data)
 #define ION_IOC_MAP _IOWR('I', 2, ion_fd_data)
 
-bool use_newer_ioctl = false;
+extern "C" {
+__attribute__((weak)) extern int ion_open();
```

Comment:
Unused? Can you make sure there are no other unused things that I haven't noticed?

Reply:
use_newer_ioctl is still used, it was just moved to the top of the file.  I haven't found any unused items.

Reply:
I'm pretty sure this particular function is unused. Ctrl+F "ion_open()" finds only this line?

Reply:
Sorry...I thought you meant the deletion of use_newer_ioctl.  All of the "__attribute__((weak)) extern" prototype declarations in this file are actually for functions that are looked up by name using dlsym.  I believe this approach was given to us by the linker team...to make things work correctly with the dynamic loader.
If you really want these prototypes removed, I can try removing them but would need to run it through all of our devices (old & new) to be sure it wasn't needed somewhere.

Reply:
All attribute((weak)) prototypes == all of the libdmabufheap.so or libion.so symbols
...and not remote_register_buf which is actually called directly in this file.

Reply:
I see now, all of these weak declarations are unused... I don't see how they could affect the generated code. Also note this is the android linker/loader, not the Hexagon one. We use this approach of dlopen/dlsym in several places and don't forward declare the functions like this. It's quite a bit of redundant specific stuff (argument types) to maintain that is dead code. I think all you need is the typedefs and function pointers.

Reply:
Change builds fine (as expected...unused functions ;-).   The generated libhalide_hexagon_host.so is also exactly the same without these prototypes...so they don't seem to be adding any meta information otherwise to the objects.   I have a feeling we may have been calling theses symbols directly at one point with the libion.so support (which was done a while back)...and switched to dlsym some point later.  The DMA-BUF support just followed the same pattern.  I'll push up another patch.

Reply:
Sorry, again didn't clean out my built objects before pushing that last patch 8-(.